doc/tutorials/content/custom_executor.rst

```diff
@@ -0,0 +1,157 @@
+.. _custom_executor:
+
+Creating Custom Executors
+--------------------------------------------------
+
+Since executors have a standard interface, you can create your own executors
+and use them.
+PCL offers support for a few executors, namely:
+
+1. Inline Executor
+2. SSE Executor
+3. OMP Executor
+4. CUDA Executor
+
+You can create specialized version of these executors i.e. executors derived from
+the executors supported by PCL, that add some additional functionality.
+
+In this tutorial we will learn how to create an executor derived from OMP executor,
+that measures and reports the time taken by a functor filter to execute.
+
+.. note::
+   This is an advanced topic and requires some knowledge on working of executors
+   and its implementation in PCL.
+
+The code
+--------
+
+First, create a file, let's say, ``custom_executor.cpp``, and place the following inside it:
+
+.. literalinclude:: sources/custom_executor/custom_executor.cpp
+   :language: cpp
+   :linenos:
+
+The explanation
+---------------
+
+Now, let's break down the code piece by piece.
+
+Here, we forward declare our custom executor struct called `omp_benchmark_executor`.
+This is required se the trait `is_executor_available` declared in the next few lines,
```

Comment:
> This is required se the trait

weird `se` word. 

If it turn out the forward declaration is not required, don't forget to remove this.

Reply:
Fixed. The declaration is required.